internal/certs/cert_manager.go

```diff
@@ -0,0 +1,183 @@
+package certs
+
+import (
+	"crypto/x509"
+	"io/ioutil"
+	"os"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/oathkeeper/driver/configuration"
+)
+
+type CertManager struct {
```

Comment:
Could you please add some tests to the certmanager? Also, what should we take care of here. For example, test different cert pools? How would this look like on Windows? :)

I haven't worked with x509 cert pools before. We should also generally ensure that:

1. The cache is used
2. The cache is invalidated properly
3. No untrusted certs are being used
4. The OS cert store is being used appropriately on the different OS'es (could be difficult for mac as we don't have a pipeline for that...)
5.